rdrf/rdrf/auth/views.py

```diff
@@ -243,3 +248,50 @@ def _get_users_verification_data(user):
         'last_name': user.last_name,
         'date_of_birth': dob,
     }
+
+
+class DeviceCookieLoginView(tfv.core.LoginView):
+
+    def get_prefix(self, request, *args, **kwargs):
+        return 'login_view'
+
+    def generate_cookie(self, user_obj):
+        utc_now = datetime.datetime.utcnow()
+        expiration_dt = utc_now + datetime.timedelta(seconds=settings.DEVICE_COOKIE_MAX_AGE)
+        existing = DeviceCookie.objects.filter(
+            user=user_obj, expires__lte=expiration_dt, expires__gte=utc_now
+        ).first()
+        expires = datetime.datetime.strftime(existing.expires if existing else expiration_dt, "%a, %d-%b-%Y %H:%M:%S GMT")
+
+        if existing:
+            existing.locked_out = False
+            existing.save()
+            encoded = existing.cookie
+        else:
+            data = {
+                'user': user_obj.username,
+                'expires': expires,
+                'max_age': settings.DEVICE_COOKIE_MAX_AGE
+            }
+            encoded = jwt.encode(data, settings.JWT_SECRET_KEY, algorithm='HS256')
+            DeviceCookie.objects.create(
+                user=user_obj,
+                max_age=settings.DEVICE_COOKIE_MAX_AGE,
+                expires=expiration_dt,
+                cookie=encoded
+            )
+
+        return encoded, expires
+
+    def done(self, form_list, **kwargs):
+        response = super().done(form_list, **kwargs)
+        cookie_value, expires = self.generate_cookie(self.get_user())
+        response.set_cookie(
+            settings.DEVICE_COOKIE_NAME,
+            cookie_value,
+            expires=expires,
+            max_age=settings.DEVICE_COOKIE_MAX_AGE,
+            domain=settings.SESSION_COOKIE_DOMAIN,
+            secure=settings.SESSION_COOKIE_SECURE or None
```

Comment:
separate setting please `DEVICE_COOKIE_DOMAIN` and `DEVICE_COOKIE_SECURE`, but in `settings.py` you can default them to the `SESSION_COOKIE_` ones.